File: languages/dotnet/src/model/DataReader.cs
using System;
using System.Collections.Generic;

namespace experimento.src.model {
 
    /// <summary>
    /// Interface para leitura de dados
    /// </summary>
    public interface DataReader {      
 
        /// <summary>
        /// Obtem todos os dados disponiveis
        /// </summary>
        /// <returns>Lista contendo todos os dados disponiveis</returns>
        List<UserInfo> read();

        /// <summary>
        /// Obtem os dados disponiveis dentro de um intervalo
        /// </summary>
        /// <param name="startIndex">Inicio do intervalo</param>
        /// <param name="endIndex">Fim do intervalo</param>
        /// <returns>Lista contendo todos os dados disponiveis dentro do intervalo especificado</returns>
        List<UserInfo> read(int startIndex, int endIndex);

    }
}

File: languages/scala/src/model/DataReader.scala
package model

/**
  * Interface para leitura de dados
  */
trait DataReader {

  /**
    * Obtem todos os dados disponiveis
    * @return Lista contendo todos os dados disponiveis
    */
  def read(): List[UserInfo]

  /**
    * Obtem os dados disponiveis dentro de um intervalo
    * @param startIndex Inicio do intervalo
    * @param endIndex Fim do intervalo
    * @return Lista contendo todos os dados disponiveis dentro do intervalo especificado
    */
  def read(startIndex: Int, endIndex: Int): List[UserInfo]
}
